SplunkADXForwarder/app.py

```diff
@@ -0,0 +1,247 @@
+                    # ingest from file
+                    df = pd.read_csv(file_path)
+                    result = ingestion_client.ingest_from_dataframe(df, ingestion_properties=ingest_props)
+                    #if ingestion result is success or queued, then move the file to temp folder
+                    if result.status == IngestionStatus.QUEUED or result.status == IngestionStatus.SUCCESS:
+                        try:
+                            move_files_to_temp_dir(file_path)
+                            logging.info("File posted to ADX and Moved file to temp folder {}".format(file_path))
+                        except Exception as err:
+                            #if any error, continue, the file will be picked up in the next event invocation
+                            logging.error("Error while moving file to temp folder {}".format(err))
+                            continue
+                    else:
+                        logging.error("File ingestion failed, will retry ingestion in the next attempt {}".format(file_path))
+                        continue         
+        else:
+            logging.error(f"The folder {buffer_folder_name} doesnt exist")
+
+def move_files_to_temp_dir(source_file_path):
+        if not os.path.isdir(os.path.join(tempfile.gettempdir(), "ADX_Ingest")):
+            temp_dir_created = tempfile.mkdtemp()
+            temp_dir = os.path.join(os.path.dirname(temp_dir_created), "ADX_Ingest")
+            os.rename(temp_dir_created, temp_dir)
+        else:
+            temp_dir = os.path.join(tempfile.gettempdir(), "ADX_Ingest")
+        if os.path.isfile(os.path.join(temp_dir, os.path.basename(source_file_path))):
+            new_file_name = get_new_file_name(os.path.basename(source_file_path))
+            shutil.move(source_file_path, os.path.join(temp_dir, new_file_name))
+        else:
+            shutil.move(source_file_path, temp_dir)
+
+
+def get_new_file_name(file_name):
+        base_name, ext = os.path.splitext(file_name)
+        counter = 1
+        new_file_name = f"{base_name}_{counter}{ext}"
+        while os.path.exists(os.path.join(new_file_name)):
+            counter += 1
+            new_file_name = f"{base_name}_{counter}{ext}"
+        return new_file_name            
+
+def get_files_not_modified(directory):
+        current_time = time.time()
+        files = []
+        for file_name in os.listdir(directory):
+            file_path = os.path.join(directory, file_name)
+            # Check if it is a regular file
+            if os.path.isfile(file_path):
+                # Get the file's modification time
+                modification_time = os.path.getmtime(file_path)
+                # Compare the modification time with the current time
+                if modification_time < current_time:
+                    files.append(file_path)
+        return files
+
+def is_file_not_being_modified(file_path):
```

Comment:
Is this for a time based flush we have this logic of not modified ?

Not that this is bad, some libs will help us cut down code and edge cases

e.g. https://pythonhosted.org/watchdog/api.html#module-watchdog.events

Reply:
Since the sample code for modulewatchdog to find the file is currently not modified was using the same timestamp, thought of using the existing psutil lib.